solver/src/settlement.rs

```diff
@@ -280,20 +295,37 @@ fn sell_order_surplus(
     buy_token_price: &BigRational,
     sell_amount_limit: &BigRational,
     buy_amount_limit: &BigRational,
-    executed_amount: &BigRational,
+    executed_sell_amount: &BigRational,
 ) -> Option<BigRational> {
     if sell_amount_limit.is_zero() {
         return None;
     }
-    let res = executed_amount * sell_token_price
-        - (executed_amount * buy_amount_limit / sell_amount_limit * buy_token_price);
+    let limit_buy_amount = executed_sell_amount * buy_amount_limit / sell_amount_limit;
+    let res = (executed_sell_amount * sell_token_price) - (limit_buy_amount * buy_token_price);
     if res.is_negative() {
         None
     } else {
         Some(res)
     }
 }
 
+fn surplus_ratio(
+    sell_token_price: &BigRational,
+    buy_token_price: &BigRational,
+    sell_amount_limit: &BigRational,
+    buy_amount_limit: &BigRational,
+) -> Option<BigRational> {
+    if buy_token_price.is_zero() || buy_token_price.is_zero() {
```

Comment:
Just to be safe I think we should also check for the limit amounts being 0. Not sure if we currently accept that in the api but it could happen in the future.

Reply:
It should be fine to ensure the denominator values are zero.